File: PG/[PG]숫자의표현.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(int n) {
    int answer = 0;
    
    for(int i =1; i <= n/2; i++){
        int sum = 0;
        int j = i;
        while(1){
            sum += j;
            if(sum > n) break;
            else if(sum == n){
                answer++;
                break;
            }
            j++;
        }        
    }
    
    return answer+1;
}

File: PG/[PG]숫자의표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;

        int sum = 0;
        int left =1, right = 1;
        while(right <= n+1){
            if(sum < n){
                sum += right++;
            }
            else if(sum == n){
                answer++;
                sum += right++;
            }
            else{
                sum -= left++;
            }
        }
        return answer;
    }
}